core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -161,6 +160,22 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     new HistoryServerDiskManager(conf, path, listing, clock)
   }
 
+  // Used to store the paths, which are being processed. This enable the replay log tasks execute
+  // asynchronously and make sure that checkForLogs would not process a path repeatedly.
+  private val processing = ConcurrentHashMap.newKeySet[String]
+
+  private def isProcessing(path: Path): Boolean = {
+    processing.contains(path.getName)
+  }
+
+  private def processing(path: Path): Unit = {
```

Comment:
Looks like the name needs to be changed - `processing(String)` would "retrieve" value, whereas `processing(Path)` would "add" the value - same name on different operations. 

Maybe `addProcessing` or better naming? Even I guess it might be better if we have a new class to extract such functionality out of FsHistoryProvider and composite it, though I agree it's too simple to extract out.

Reply:
thanks for you suggestion, the name was referred that of `blacklist`.
I think the function of `processing` is similar with `blacklist`, so how about just keeping consistent with `blacklist`?
https://github.com/apache/spark/blob/21db2f86f7c196c535e8f0b5675ae48cb2c372f7/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L164-L173

Reply:
Ah OK didn't recognize it. Looks OK to me.